File: Swift-C++-Testsuite/test/stringsAndCharacters/Swift/StringIndices.swift
@main
struct MyMain {

    static func main() {

        var greeting = "Guten Tag!"
        greeting[greeting.startIndex]
        // G
        greeting[greeting.index(before: greeting.endIndex)]
        // !
        greeting[greeting.index(after: greeting.startIndex)]
        // u
        let index = greeting.index(greeting.startIndex, offsetBy: 7)
        greeting[index]
        // a

        for index in greeting.indices {
            var curr = greeting[index]
        }

        greeting.insert("!", at: greeting.endIndex)

        greeting.removeSubrange(greeting.index(greeting.endIndex, offsetBy: -2)..<greeting.endIndex)
    }
}

File: Swift-C++-Testsuite/test/stringsAndCharacters/C++/StringIndices.cpp
#include <string>

int main() {
    std::string greeting = "Guten Tag!";
    char beginning = greeting[0];
    char beginningAlt = *greeting.begin();

    char end = *greeting.end();
    char endAlt = greeting[greeting.length() -1];

    std::string::iterator offset7 = greeting.begin() + 7;
    char offsetChar = *offset7;

    for ( std::string::iterator it=greeting.begin(); it!=greeting.end(); ++it) {
        char curr = *it;
    }

    greeting.insert(greeting.length()-1, "!");

    greeting.erase(greeting.length()-2, greeting.length()-1);
}
